### ISSUE TITLE
Ideas for improvement

### ISSUE THREAD

**Issue by jdewfiez** [sky-music/sky-music-website-project#9]
I was thinking about how the website could be better (I guess). So here is my suggestion:
- The form should be done in 1 page, and not multiple pages. 
- ^ From this, we can assign different ID/UUID for each user and store them in user's browser cookie.
- ^ Also sheet editing will be possible.
- We can move to using 3rd party storage for bigger database. (might not recommend right now)
- The result page should render the HTML instead of the download page, and put some download link below the HTML form.

What do you guys think about those idea? I currently have zero experience in Django and still learning it.

**Comment by jmmelko** [comment]
Well yes, or course these are very good proposals. I know that the process of clicking so much and having one page at a time is tiring.

Here are some answers:

* Actually the music sheet maker is a linear program, meaning that each question is generated (customized) using the answer you gave to previous questions.
What I did is just a website calling our python program before displaying each page to prompt questions on the screen. Yes, I know, that is not efficient.
If everything is on a single page (as envisioned initially by Tracey), this would require some additional python coding and possibly JavaScript scripting made inside the website. This requires manpower I am not able to supply. In addition, this means that the website code will have to maintain separately. I mean : bugs solved In the python script will not be automatically ported to the website.

* the UUID things is already in place

* about the tip, yes, you are right

* yes we should render the HTML directly, the problem is that the HTML is generated by the python script, not the website code, so unless we use a trick the website would have to parse the HTML code and modify it on the fly before displaying it. It is doable, but I won’t do it myself.

**Comment by jdewfiez** [comment]
Well yes, you are right. We have to port the question/instructions into JS and then parse the answer using `POST` request into the program itself. Something I _kinda_ have the ability to do, but I still don't know how the program proceed with the user's answer.

And oh, I meant the UUID is something like `w7AS9k` and not ascending number for each question.

**Comment by t1-tracey** [comment]
Yeah unfortunately we don’t have enough resources to focus on more than one platform at a time, right now. Feel free to see what you can do though^^
